proxy/src/lib.rs

```diff
@@ -314,27 +315,55 @@ impl Proxy {
         Ok(table)
     }
 
-    async fn maybe_open_partition_table_if_not_exist(
+    async fn get_partition_table_info(
         &self,
         catalog_name: &str,
         schema_name: &str,
-        table_name: &str,
-    ) -> Result<()> {
-        if let Err(e) = self
-            .open_partition_table_inner(catalog_name, schema_name, table_name)
-            .await
-        {
-            warn!("Open partition table failed, err:{e:?}");
+        base_name: &str,
+        part_info: &PartitionInfo,
+    ) -> Result<TableInfo> {
+        let get_inner = |i| async move {
+            let sub_partition_table_name = util::get_sub_partition_name(base_name, part_info, i);
+            let table = self
+                .instance
+                .remote_engine_ref
+                .get_table_info(GetTableInfoRequest {
+                    table: TableIdentifier {
+                        catalog: catalog_name.to_string(),
+                        schema: schema_name.to_string(),
+                        table: sub_partition_table_name,
+                    },
+                })
+                .await
+                .box_err()
+                .with_context(|| ErrWithCause {
+                    code: StatusCode::INTERNAL_SERVER_ERROR,
+                    msg: "Failed to get table",
+                })?;
+
+            Ok(table)
+        };
+
+        let part_num = part_info.get_definition_num();
+        if part_num == 1 {
```

Comment:
Do not check the `partition number == 1`, because it is very likely that it will not be achieved.